framework/src/executioners/TransientBase.C

```diff
@@ -347,7 +347,15 @@ TransientBase::incrementStepOrReject()
       _time_old = _time;
       _t_step++;
 
-      _problem.advanceState();
+      bool advance_state = true;
+      const auto & tis = getTimeIntegrators();
+      for (auto & ti : tis)
+        // We do not want to advance state all if this is an explicit time integrator
+        if (ti->controlsState())
+          advance_state = false;
+
+      if (advance_state)
+        _problem.advanceState();
```

Comment:
I wonder if we shouldn't change this to
```suggestion
      const auto & tis = getTimeIntegrators();
      mooseAssert(tis.size() == 1, "Expected exactly one time integrator.");
      if (!tis[0]->controlsState())
        _problem.advanceState();
```

Reply:
Your version isn't exactly future proof, as we'd need to figure out what to do when we have a mix of explicit and implicit TIs.

Reply:
So for the first part we don't expect one time integrator currently. We can already have multiple time integrators. But we currently can only 1 time integrator that has `overridesSolve` equal to true. We have a check `NonlinearSystem` that prevents having multiple time integrators when a single one overrides the solve. 
Agreed not future proof though.

Reply:
```suggestion
      bool advance_problem_state = true;
      const auto & tis = getTimeIntegrators();
      for (auto & ti : tis)
        // We do not want to advance the problem state here if a time integrator is already doing so itself
        if (ti->advancesProblemState())
        {
          advance_problem_state = false;
          break;
        }

      if (advance_problem_state)
        _problem.advanceState();
      else if (tis.size() > 1)
        mooseError("Either there must be a single time integrator which advances state or none of the time integrators should advance state.");
```